core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/ExpressionParser.scala

```diff
@@ -1,29 +1,28 @@
 package org.broadinstitute.dsde.rawls.expressions
 
-import org.broadinstitute.dsde.rawls.model.{AttributeString, MethodConfiguration, ValidatedMethodConfiguration}
+import org.broadinstitute.dsde.rawls.model.{AttributeString, ParsedMCExpressions}
 
 import scala.util.{Failure, Success, Try}
-
 import spray.json._
 
 // a thin abstraction layer over SlickExpressionParser
 
 object ExpressionParser {
-  def parseMCExpressions(methodConfiguration: MethodConfiguration, parser: SlickExpressionParser): ValidatedMethodConfiguration = {
+  def parseMCExpressions(inputs: Map[String, AttributeString], outputs: Map[String, AttributeString], parser: SlickExpressionParser): ParsedMCExpressions = {
     def parseAndPartition(m: Map[String, AttributeString], parseFunc:String => Try[Unit] ) = {
       val parsed = m map { case (key, attr) => (key, parseFunc(attr.value)) }
       ( parsed collect { case (key, Success(_)) => key } toSeq,
         parsed collect { case (key, Failure(regret)) => (key, regret.getMessage) } )
     }
-    val (successInputs, failedInputs)   = parseAndPartition(methodConfiguration.inputs, parseAttributeExpr(parser) )
-    val (successOutputs, failedOutputs) = parseAndPartition(methodConfiguration.outputs, parseOutputExpr(parser) )
+    val (successInputs, failedInputs)   = parseAndPartition(inputs, parseAttributeExpr(parser) )
+    val (successOutputs, failedOutputs) = parseAndPartition(outputs, parseOutputExpr(parser) )
 
-    ValidatedMethodConfiguration(methodConfiguration, successInputs, failedInputs, successOutputs, failedOutputs)
+    ParsedMCExpressions(successInputs, failedInputs, successOutputs, failedOutputs)
   }
 
   private def parseAttributeExpr(parser: SlickExpressionParser)(expression: String): Try[Unit] = {
     Try(expression.parseJson) match {
-      case Success(parsed) => Success(true)
+      case Success(parsed) => Success(())
       case Failure(regret) => parser.parseAttributeExpr(expression) map (_ => ())
```

Comment:
A shorthand for `map(_ => ())` is `.void`. E.g.:
```
parser.parseAttributeExpr(expression)
```
It needs the following imports:
```
import cats.syntax.functor._   // adds the .void syntax
import cats.instances.try_._    // the thing you're trying to void (could be future, option, etc) 
```

Reply:
Aha!  I wanted that but I couldn't figure out the right imports.